br-testing-automation/auto-br.sh

```diff
@@ -0,0 +1,289 @@
+        
+        info "Editing backup yaml..."
+        sed -i -E "s/<backup storage location name>/$BACKUP_STORAGE_LOCATION_NAME/" ./templates/sf-backup.yaml
+        sed -i -E "s/<application name>/$APPLICATION/" ./templates/sf-backup.yaml
+        sed -i -E "s/<backup policy name>/$BACKUP_POLICY/" ./templates/sf-backup.yaml
+        sed -i -E "s/<backup name>/$BACKUP_NAME/" ./templates/sf-backup.yaml
+        if [[ $clustertype == "spoke" ]]; then
+            sed -i -E "s/<Cluster CR name only for backups on spoke cluster>/$TARGET_CLUSTER/" ./templates/sf-backup.yaml
+        else
+            ${YQ} -i 'del(.spec.appCluster)' ./templates/sf-backup.yaml || error "Could not remove appCluster field from backup yaml."
+        fi
+        if [[ $SF_NAMESPACE != "ibm-spectrum-fusion-ns" ]]; then
+            ${YQ} -i '.metadata.namesace = "'${SF_NAMESPACE}'"' ./templates/sf-backup.yaml || error "Could not update namespace value to $SF_NAMESPACE in backup yaml."
+        fi
+
+        ${OC} apply -f ./templates/sf-backup.yaml || error "Failed to apply backup yaml."
+        success "Backup $BACKUP_NAME successfully applied on hub server $HUB_SERVER to backup target cluster $TARGET_CLUSTER"
+        
+    fi
+    
+    if [[ $brtype == "restore" ]]; then
+        cp ../velero/spectrum-fusion/templates/sf-restore.yaml ./templates/sf-restore.yaml
+
+        info "Editing restore yaml..."
+        sed -i -E "s/<backup storage location name>/$BACKUP_STORAGE_LOCATION_NAME/" ./templates/sf-restore.yaml
+        sed -i -E "s/<application name>/$APPLICATION/" ./templates/sf-restore.yaml
+        sed -i -E "s/<backup policy name>/$BACKUP_POLICY/" ./templates/sf-restore.yaml
+        sed -i -E "s/<restore name>/$RESTORE_NAME/" ./templates/sf-restore.yaml
+        sed -i -E "s/<backup name>/$BACKUP_NAME/" ./templates/sf-restore.yaml
+
+        if [[ $clustertype == "spoke" ]]; then
+            sed -i -E "s/<Cluster CR name only for restores to spoke cluster>/$TARGET_CLUSTER/" ./templates/sf-restore.yaml
+        else
+            ${YQ} -i 'del(.spec.targetCluster)' ./templates/sf-restore.yaml || error "Could not remove targetCluster field from restore yaml."
+        fi
+        if [[ $SF_NAMESPACE != "ibm-spectrum-fusion-ns" ]]; then
+            ${YQ} -i '.metadata.namesace = "'${SF_NAMESPACE}'"' ./templates/sf-restore.yaml || error "Could not update namespace value to $SF_NAMESPACE in restore yaml."
+        fi
+
+        ${OC} apply -f ./templates/sf-restore.yaml || error "Failed to apply restore yaml."
+        success "Restore $RESTORE_NAME successfully applied on hub server $HUB_SERVER to restore target cluster $TARGET_CLUSTER"
+    fi
+}
+
+function wait_for_br(){
+    type=$1
+    resource_name=$2
+    retries=$3
+    time=30
+    title "Waiting for $type $resource_name to complete..."
+    status=$(${OC} get $type $resource_name -n $SF_NAMESPACE -o yaml | ${YQ} '.status.phase')
```

Comment:
Just a tiny comment, if we could use `kind.group` to get the resource, it would prevent the script from failing to find the exact Backup resource we're looking for in case there are multiple resources with the same kind name in the cluster.

Here the cmd I am using:  
```
status=$(${OC} get backup.data-protection.isf.ibm.com $resource_name -n $SF_NAMESPACE -o yaml | ${YQ} '.status.phase')
```

Reply:
great sugggestion @YCShen1010